### PR TITLE
feat(`server`): lazy `rawInput` getter

### PR DESCRIPTION
**Pull request by KATT**
Closes #

## 🎯 Changes

Make input a lazy getter to prepare for better content type support

### CODE REVIEW

File: packages/server/src/core/middleware.ts [hunk 1]
```diff
@@ -230,13 +233,11 @@ function isPlainObject(obj: unknown) {
  * Please note, `trpc-openapi` uses this function.
  */
 export function createInputMiddleware<TInput>(parse: ParseFn<TInput>) {
-  const inputMiddleware: ProcedureBuilderMiddleware = async ({
-    next,
-    rawInput,
-    input,
-  }) => {
+  const inputMiddleware: ProcedureBuilderMiddleware = async (opts) => {
```
Comment (Nick-Lucas) [inline packages/server/src/core/middleware.ts:236]: I can't believe you found another destructure during this, that's a throwback!

File: .wip/changes.md [hunk 2]
```diff
@@ -17,3 +17,5 @@
 Check their migration guide: https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5
 
 FIXME: Update our docs where applicable
+
+## `rawInput` in middleware is now a `getRawInput`
```
Comment (Nick-Lucas) [inline .wip/changes.md:21]:
```suggestion
## `rawInput` in middleware is now a `getRawInput`

While we're not doing anything differently internally (just yet) this is help support a much requested feature in tRPC: content types other than JSON.
```

Some blip of detail on why the change is worth it maybe? IDK

**Review by Nick-Lucas — COMMENTED** [review]
Looks good, 1 possible defect

File: packages/server/src/core/middleware.ts [hunk 3]
```diff
@@ -230,13 +233,11 @@ function isPlainObject(obj: unknown) {
...
     let parsedInput: ReturnType<typeof parse>;
+
     try {
+      const rawInput = await opts.getRawInput();
```
[the remark this answers is not in the document]
Comment (KATT) [reply packages/server/src/core/middleware.ts:240]: Good thinking
